Authentication.md

```diff
@@ -0,0 +1,349 @@
+### Programming example
+```go
+import {
+    "github.com/IBM/go-sdk-core/core"
+    "<appropriate-git-repo-url>/exampleservicev1"
+}
+...
+// Create the authenticator.
+authenticator := &core.CloudPakForDataAuthenticator{
+    Username: "myuser",
+    Password: "mypassword",
+    URL: "https://mycp4dhost.com/",
+}
+
+// Create the service options struct.
+options := &exampleservicev1.ExampleServiceV1Options{
+    Authenticator: authenticator,
+}
+
+// Construct the service instance.
+service := exampleservicev1.NewExampleServiceV1(options)
+
+// 'service' can now be used to invoke operations.
+```
+### Configuration example
+External configuration:
+```
+export EXAMPLE_SERVICE_AUTH_TYPE=cp4d
+export EXAMPLE_SERVICE_USERNAME=myuser
+export EXAMPLE_SERVICE_PASSWORD=mypassword
+export EXAMPLE_SERVICE_URL=https://mycp4dhost.com/
+```
+Application code:
+```go
+import {
+    "github.com/IBM/go-sdk-core/core"
+    "<appropriate-git-repo-url>/exampleservicev1"
+}
+...
+// Construct the authenticator from external configuration information for service "example_service".
+authenticator := &core.GetAuthenticatorFromEnvironment("example_service")
+
+// Create the service options struct.
+options := &exampleservicev1.ExampleServiceV1Options{
+    Authenticator: authenticator,
+}
+
+// Construct the service instance.
+service := exampleservicev1.NewExampleServiceV1(options)
+
+// 'service' can now be used to invoke operations.
+```
+## No Auth Authentication
+The `NoAuthAuthenticator` is a placeholder authenticator which performs no actual authentication function.   It can be used in situations where authentication needs to be bypassed, perhaps while developing or debugging an application or service.
+### Properties
+None
+### Programming example
+```go
+import {
+    "github.com/IBM/go-sdk-core/core"
+    "<appropriate-git-repo-url>/exampleservicev1"
+}
+...
+// Create the authenticator.
+authenticator := &core.NoAuthAuthenticator{}
+
+// Create the service options struct.
+options := &exampleservicev1.ExampleServiceV1Options{
+    Authenticator: authenticator,
+}
+
+// Construct the service instance.
+service := exampleservicev1.NewExampleServiceV1(options)
+
+// 'service' can now be used to invoke operations.
+```
+### Configuration example
+External configuration:
+```
+export EXAMPLE_SERVICE_AUTH_TYPE=noauth
+```
+Application code:
+```go
+import {
+    "github.com/IBM/go-sdk-core/core"
+    "<appropriate-git-repo-url>/exampleservicev1"
+}
+...
+// Construct the authenticator from external configuration information for service "example_service".
+authenticator := &core.GetAuthenticatorFromEnvironment("example_service")
+
+// Create the service options struct.
+options := &exampleservicev1.ExampleServiceV1Options{
+    Authenticator: authenticator,
+}
+
+// Construct the service instance.
+service := exampleservicev1.NewExampleServiceV1(options)
+
+// 'service' can now be used to invoke operations.
+```
```

Comment:
I wonder if this shouldn't be `AUTHENTICATION.md` - I feel like most markdown files use all caps. Really not a big deal though, just thinking out loud